app/models/post.rb

```diff
@@ -332,4 +338,34 @@ def invalidate_caches
     return unless saved_change_to_authors_locked?
     Post::Author.clear_cache_for(authors)
   end
+
+  def create_written
+    return if skip_written
+    self.written = Reply.create!(
+      post: self,
+      reply_order: 0,
+      user: user,
+      content: content,
+      icon: icon,
+      character: character,
+      character_alias: character_alias,
+      created_at: created_at,
+      updated_at: edited_at,
+      skip_regenerate: true,
```

Comment:
Do we need to worry about `skip_notify`, `is_import`, or `skip_post_update`? (I'm guessing yes, especially on import posts, but not 100% sure.)

Reply:
skip_notify seems redundant, since this'll always have the same user as the post, but I'll throw the other two on.